### ISSUE TITLE
Consumable, Service or Stockable products ?

### ISSUE THREAD

**Issue by hparfr** [OCA/product-pack#11]
Hi, 
I'm looking about these promising modules and one question arise:

What type should be the parent product (is_pack = true) ?

If service or consumable, stock_product_pack seems useless : the smart buttons don't appear).
If stockable, then inventory adjustments or "update quantity" button is present but useless and it's in the transferts it's never reserved so it ask for backorder creation.

**Comment by pedrobaeza** [comment]
It should storable, but details like the "Update quantity" visibility hasn't been taking into account. With module `stock_product_pack`, the quantity on hand should be correctly computed and reserved AFAIK.

**Comment by hparfr** [comment]
Thanks for this precision.

I confirm with sale_stock_product_pack the quantity is right., but the parent product (the pack) is not reserved.

**Comment by pedrobaeza** [comment]
@ernestotejeda can you check what @hparfr mentions?

**Comment by hparfr** [comment]
In v12.
product: name: "element of pack",  type product, pack_ok = false; update qty to some items
product: name:"pack" , type product , pack_ok = true, pack lines: "elements of pack", pack_type: detailled



![Capture d’écran 2020-03-05 à 14 42 19-fullpage](https://user-images.githubusercontent.com/8331206/75987425-330eee80-5ef0-11ea-9657-1b57a19c69d7.png)

In a SO, add product pack, confirm.
In the transfert the quantity for "pack" is never reserved, the quantity for "element of pack" is well reserved.


![Capture d’écran 2020-03-05 à 14 42 37-fullpage](https://user-images.githubusercontent.com/8331206/75987444-399d6600-5ef0-11ea-9d14-8abf1229632c.png)
When you transfert, it asks for a backorder.

**Comment by FedericoLaTorre** [comment]
Where can I find the module sale_stock_product_pack?

**Comment by pedrobaeza** [comment]
I meant https://github.com/OCA/product-pack/tree/12.0/stock_product_pack

**Comment by FedericoLaTorre** [comment]
Thanks, so I have already installed that module, but I have the same issue as user hparfr. Quantity is never reserved, asked for backorder and pack products are never deducted from stock.

**Comment by pedrobaeza** [comment]
Yes, that's currently a missing feature as we have checked. The solution for now is to put pack as consumable instead of stockable.

**Comment by infoitservit** [comment]
Currently this module is a "must" for an ecommerce, its importance is vital to compete with magento and also other platforms.
I tried many configurations, in anyone there is a problem:
1) Setting it as stockable creates problems on the picking step as underlined before by @FedericoLaTorre
2) Setting it as consumable makes impossibile to manage the right inventory, so one could buy undefined quantity even if one of the subproducts is not available
3) I found that setting it as "Pack Component Price" = "Totalized in main product", on the ecommerce, when added to cart, it correctly calculate price for the pack only but the, clicking on "process checkout", odoo refresh the line and add the price of the single subproducts

![image](https://user-images.githubusercontent.com/3064689/92144946-c2ea1280-ee17-11ea-9e86-ddc164ee5b79.png)
![image](https://user-images.githubusercontent.com/3064689/92145183-22482280-ee18-11ea-8c77-46ea58c897d0.png)

4) When setted up as "Not Detailed", after an order placed on the website, Odoo will never show subproducts, even in the picking phase (where it is really important that they appear)

![image](https://user-images.githubusercontent.com/3064689/92145573-b4e8c180-ee18-11ea-9655-db2f403f6544.png)
![image](https://user-images.githubusercontent.com/3064689/92145682-dea1e880-ee18-11ea-8914-d4b99c72634b.png)

I would like to help to develop a complete and robust module, similar to the one present on professional ecommerce platforms like Magento from which i came from.
There is a way to finance this module to be sure that in a fixed time it will be released stable as it should be?
